fmt/printf.h

```diff
@@ -588,6 +600,8 @@ FMT_VARIADIC(int, printf, CStringRef)
   \endrst
  */
 inline int fprintf(std::ostream &os, CStringRef format_str, ArgList args) {
+  std::mutex mtx;
+  std::unique_lock<std::mutex> lck(mtx);
```

Comment:
I think the library shouldn't provide weaker thread-safety guarantees than the underlying I/O API, that's all. If `ostream` is thread-safe (as in http://open-std.org/JTC1/SC22/WG21/docs/papers/2015/p0053r0.html) then the function that uses it should be too. But I strongly believe that explicitly synchronizing access to `ostream` is out of scope of the library and if anyone is interested, it provides a simple way to write a custom formatting function in just a few lines of code. This function can take a mutex, or even better, one can put `ostream` and `mutex` in a class and provide synchronized formatting methods.